Cargo.toml

```diff
@@ -70,11 +70,16 @@ members = [
 ]
 exclude = ["benches", "distr_test"]
 
+# Force chacha20 to use the local rand_code to avoid compiling two different "rand_core"s.
+# This is necessary even if the specified versions are the same.
+[patch.crates-io]
+rand_core = { path = "rand_core" }
+
 [dependencies]
 rand_core = { path = "rand_core", version = "0.9.0", default-features = false }
 log = { version = "0.4.4", optional = true }
 serde = { version = "1.0.103", features = ["derive"], optional = true }
-rand_chacha = { path = "rand_chacha", version = "0.9.0", default-features = false, optional = true }
+chacha20 = { version = "=0.10.0-rc.1", default-features = false, features = ["rng", "legacy"], optional = true }
```

Comment:
We can depend on the "legacy" feature for now, but add a `TODO` comment to remove it on the next version.